lapis2/src/main/kotlin/org/genspectrum/lapis/response/SiloResponse.kt

```diff
@@ -8,11 +8,38 @@ import com.fasterxml.jackson.databind.JsonNode
 import com.fasterxml.jackson.databind.JsonSerializer
 import com.fasterxml.jackson.databind.SerializerProvider
 import io.swagger.v3.oas.annotations.media.Schema
+import org.genspectrum.lapis.controller.CsvRecord
 import org.springframework.boot.jackson.JsonComponent
 
 const val COUNT_PROPERTY = "count"
 
-data class AggregationData(val count: Int, @Schema(hidden = true) val fields: Map<String, JsonNode>)
+data class AggregationData(val count: Int, @Schema(hidden = true) val fields: Map<String, JsonNode>) : CsvRecord {
+    override fun asArray() = fields.values.map { it.asText() }.plus(count.toString()).toTypedArray()
+    override fun asHeader() = fields.keys.plus(COUNT_PROPERTY).toTypedArray()
+}
+
+data class DetailsData(val map: Map<String, JsonNode>) : Map<String, JsonNode> by map, CsvRecord {
+    override fun asArray() = values.map { it.asText() }.toTypedArray()
+    override fun asHeader() = keys.toTypedArray()
+}
+
+@JsonComponent
+class DetailsDataSerializer : JsonSerializer<DetailsData>() {
+    override fun serialize(value: DetailsData, gen: JsonGenerator, serializers: SerializerProvider) {
+        gen.writeStartObject()
+        value.forEach { (key, value) -> gen.writeObjectField(key, value) }
+        gen.writeEndObject()
+    }
+}
+
+@JsonComponent
+class DetailsDataDeserializer : JsonDeserializer<DetailsData>() {
+    override fun deserialize(p: JsonParser, ctxt: DeserializationContext): DetailsData {
+        val node = p.readValueAsTree<JsonNode>()
+        val fields = node.fields().asSequence().associate { it.key to it.value }
+        return DetailsData(fields)
```

Comment:
I would have expected something in the spirit of (since it worked without an explicit deserializer before, too):
```suggestion
        return DetailsData(p.readValueAsTree<Map<String, JsonNode>>())
```

Reply:
Good idea. The readValueAsTree can not get the Map directly. One need to use 
`return DetailsData(p.readValueAs(object : TypeReference<Map<String, JsonNode>>() {}))`